src/algebra/big_operators.lean

```diff
@@ -338,6 +350,30 @@ by haveI := classical.dec_eq α; exact
     by rw [prod_image]; exact assume x _ y _, subtype.eq
   ... = _ : by rw [attach_image_val]
 
+/-- A product over `s.subtype p` equals one over `s.filter p`. -/
+@[to_additive "A sum over `s.subtype p` equals one over `s.filter p`."]
+lemma prod_subtype_eq_prod_filter (f : α → β) {p : α → Prop} [decidable_pred p] :
+  ∏ x in s.subtype p, f x = ∏ x in s.filter p, f x :=
```

Comment:
Perhaps this deserves `@[simp]`? It it removing a coercion.

Reply:
Added more use of `@[simp]` as suggested.